ksql-metastore/src/main/java/io/confluent/ksql/metastore/StructuredDataSource.java

```diff
@@ -21,31 +21,30 @@
 
 import io.confluent.ksql.query.QueryId;
 import io.confluent.ksql.serde.DataSource;
-import io.confluent.ksql.util.KsqlException;
 import io.confluent.ksql.util.timestamp.TimestampExtractionPolicy;
 
 public abstract class StructuredDataSource implements DataSource {
 
-  final String dataSourceName;
-  final DataSourceType dataSourceType;
-  final Schema schema;
-  final Field keyField;
-  final TimestampExtractionPolicy timestampExtractionPolicy;
+  protected final String dataSourceName;
```

Comment:
why not just leave these package private?

Reply:
protected gives better encapsulation.